cernopendata_client/cli.py

```diff
@@ -100,25 +102,51 @@ def get_file_locations(server, recid, doi, title, protocol, expand):
     if recid is not None:
         validate_recid(recid)
     record_json = get_record_as_json(server, recid, doi, title)
-    file_locations = [file["uri"] for file in record_json["metadata"]["files"]]
-    if expand:
-        # let's unwind file indexes
-        file_locations_expanded = []
-        for file in file_locations:
-            if file.endswith("_file_index.txt"):
-                url_file = file.replace("root://eospublic.cern.ch/", server)
-                req = requests.get(url_file)
-                for url_individual_file in req.text.split("\n"):
-                    if url_individual_file:
-                        file_locations_expanded.append(url_individual_file)
-            elif file.endswith("_file_index.json"):
-                pass
-            else:
-                file_locations_expanded.append(file)
-        file_locations = file_locations_expanded
-
-    if protocol == "http":
-        file_locations = [
-            f.replace("root://eospublic.cern.ch/", server) for f in file_locations
-        ]
+    file_locations = get_files_list(server, record_json, protocol, expand)
     click.echo("\n".join(file_locations))
+
+
+@cernopendata_client.command()
+@click.option("--recid", type=click.INT, help="Record ID")
+@click.option("--doi", help="Digital Object Identifier.")
+@click.option("--title", help="Record title")
+@click.option(
+    "--protocol",
+    default="root",
+    type=click.Choice(["root", "http"]),
+    help="Protocol to be used in links.",
+)
+@click.option(
+    "--expand/--no-expand", default=True, help="Expand file indexes? [default=yes]"
+)
+@click.option(
+    "--server",
+    default="http://opendata.cern.ch",
+    type=click.STRING,
+    help="Which CERN Open Data server to query? [default=http://opendata.cern.ch]",
+)
+def download_files(server, recid, doi, title, protocol, expand):
+    """Download the list of files belonging to a dataset."""
+    validate_server(server)
+    if recid is not None:
+        validate_recid(recid)
+    record_json = get_record_as_json(server, recid, doi, title)
+    file_locations = get_files_list(server, record_json, protocol, expand)
+
+    path = str(recid)
+    try:
+        os.mkdir(path)
+    except OSError:
+        print("Creation of the directory %s failed" % path)
+
+    for file_location in file_locations:
+        file_dest = path + "/" + file_location.split("/")[-1]
+        with open(file_dest, "wb") as f:
+            print("File Downloading")
```

Comment:
It would be nice to introduce some output formatting cosmetics, e.g. currently people see:

```console
$ cernopendata-client download-files --recid 5500 --protocol http
Creation of the directory 5500 failed
File Downloading
File Downloading kiB (100%)
File Downloading81 kiB (100%)
File Downloading kiB (100%)
File Downloading14 kiB (100%)
File Downloading15 kiB (100%)
File Downloading kiB (100%)
File Downloading kiB (100%)
File Downloading kiB (100%)
File Downloading kiB (100%)
File Downloading17 kiB (100%)
Downloading: 90/90 kiB (100%)
```

I would expect to see the URI or at least the file name that is being currently downloaded.  Also, the total count, so that people can estimate the ETA.  Somethhing like this:

```
==> Downloading file 1 of 10:  demoanalyzer_cfg_level3data.py
Downloading 123/123456789 KiB (1.23%)
==> Downloading file 2 of 10: demoanalyzer_cfg_level3MC.py
...
```